src/supertux/menu/editor_level_select_menu.cpp

```diff
@@ -134,16 +134,44 @@ EditorLevelSelectMenu::create_item(bool worldmap)
   }
 }
 
+void
+EditorLevelSelectMenu::open_level(const std::string& filename)
+{
+  auto editor = Editor::current();
+  editor->set_level(filename);
+  MenuManager::instance().clear_menu_stack();
+}
+
 void
 EditorLevelSelectMenu::menu_action(MenuItem& item)
 {
   auto editor = Editor::current();
   World* world = editor->get_world();
   if (item.get_id() >= 0)
   {
-    editor->set_level(m_levelset->get_level_filename(item.get_id()));
 
-    MenuManager::instance().clear_menu_stack();
+    std::string file_name = m_levelset->get_level_filename(item.get_id());
+    std::string file_name_full = FileSystem::join(editor->get_level_directory(), file_name);
+
+    if (PHYSFS_exists((file_name_full + "~").c_str())) {
+      auto dialog = std::make_unique<Dialog>(/* passive = */ false, /* auto_clear_dialogs = */ false);
+      dialog->set_text(_("An auto-save file has been found. Would you like to open the auto-save file?\n"));
```

Comment:
Not sure about the string here. I'd suggest something more user-friendly, such as "Would you like to restore your the level you were last working on?" or something? Or other suggestions?

Reply:
I'm open to changing the line if needed, but I'd like to keep the part that says the message is shown because a recovery file was found. There might be bugs I haven't noticed that would leave the recovery file there (say, the file deletion fails for some reason), and seeing the warning without knowing why it pops in the first place could cause some confusion.

Reply:
A few suggestions:
- "It looks like the editor crashed last time you opened this level. Would you like to continue where you left off?"
- "An auto-save recovery file was found. Would you like to restore the recovery file and resume where you were before the editor crashed?"
- "It looks like you have some unsaved progress on that level. This can happen if the editor crashes. Would you like to resume where you left off?"

Reply:
I'd suggest making one that is as short but effective as possible.  Players do not like to read one large string that can be easily summarized and have the same effect.